source/transactions-convenient-api/tests/commit-transienttransactionerror.yml

```diff
@@ -0,0 +1,191 @@
+            lsid: session0
+            # txnNumber is incremented when retrying the transaction
+            txnNumber: { $numberLong: "2" }
+            startTransaction: true
+            autocommit: false
+            # omitted fields
+            readConcern: ~
+            writeConcern: ~
+          command_name: insert
+          database_name: *database_name
+      -
+        command_started_event:
+          command:
+            commitTransaction: 1
+            lsid: session0
+            txnNumber: { $numberLong: "2" }
+            autocommit: false
+            # omitted fields
+            readConcern: ~
+            startTransaction: ~
+            writeConcern: ~
+          command_name: commitTransaction
+          database_name: admin
+    outcome: &outcome
+      collection:
+        data:
+          - { _id: 1 }
+  -
+    description: transaction is retried after commitTransaction TransientTransactionError (LockTimeout)
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 2 }
+      data:
+          failCommands: ["commitTransaction"]
+          errorCode: 24 # SnapshotUnavailable
+          closeConnection: false
+    operations:
+      -
+        name: withTransaction
+        object: session0
+        callback: *callback
+    expectations: *expectations
+    outcome: *outcome
+  -
+    description: transaction is retried after commitTransaction TransientTransactionError (WriteConflict)
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 2 }
+      data:
+          failCommands: ["commitTransaction"]
+          errorCode: 112 # WriteConflict
+          closeConnection: false
+    operations:
+      -
+        name: withTransaction
+        object: session0
+        callback: *callback
+    expectations: *expectations
+    outcome: *outcome
+  -
+    description: transaction is retried after commitTransaction TransientTransactionError (SnapshotUnavailable)
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 2 }
+      data:
+          failCommands: ["commitTransaction"]
+          errorCode: 246 # SnapshotUnavailable
+          closeConnection: false
+    operations:
+      -
+        name: withTransaction
+        object: session0
+        callback: *callback
+    expectations: *expectations
+    outcome: *outcome
+  -
+    description: transaction is retried after commitTransaction TransientTransactionError (NoSuchTransaction)
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 2 }
+      data:
+          failCommands: ["commitTransaction"]
+          errorCode: 251 # NoSuchTransaction
+          closeConnection: false
+    operations:
+      -
+        name: withTransaction
+        object: session0
+        callback: *callback
+    expectations: *expectations
+    outcome: *outcome
+  -
+    description: transaction is retried after commitTransaction TransientTransactionError (PreparedTransactionInProgress)
```

Comment:
What makes this one retryable?

Reply:
For `251 # NoSuchTransaction` above, the server is adding the transient TransientTransactionError label in the response, but that's not happening for this one.  Perhaps this requires a 4.2 server?  Is does seem like a distributing transaction-related error code.

Reply:
Indeed, that was introduced in 4.1.6. See https://github.com/mongodb/specifications/pull/435/commits/48d54031d7468880b65edefce9bd37746a8c3dd1